core/src/main/java/org/elasticsearch/index/shard/IndexShardRelocatedException.java

```diff
@@ -29,10 +29,14 @@
 public class IndexShardRelocatedException extends IllegalIndexShardStateException {
```

Comment:
I had another look and think we should keep it as is for this PR. I looked at other subclasses of `IllegalIndexShardStateException`, and most of them could be removed as well if we follow the argument given here.
I will leave that for another PR (where this can be discussed then).